avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1402,14 +1509,14 @@ else if (schema.type() == Schema.Type.INT32)
   private static String unionMemberFieldName(org.apache.avro.Schema schema) {
     if (schema.getType() == org.apache.avro.Schema.Type.RECORD ||
         schema.getType() == org.apache.avro.Schema.Type.ENUM) {
-      return splitName(schema.getName())[1];
+      return schema.getFullName();
```

Comment:
This also seems like a breaking change since the field names will change. These aren't necessarily deal breakers but unfortunately will need to be put behind a config flag unless we can show they don't actually cause problems.